model/src/test/kotlin/IdentifierTest.kt

```diff
@@ -172,6 +173,18 @@ class IdentifierTest : WordSpec({
 
             purl shouldBe "pkg:type/namespace/name@release%20candidate"
         }
+
+        "allow query parameters" {
+            val purl = createPurl(
+                "type",
+                "namespace",
+                "name",
+                "version",
+                mapOf("argName" to "argValue")
```

Comment:
Could you test with a map containing at least two parameters? So the different separator characters in the query string will be tested.